streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -649,19 +434,38 @@ private void processCreateTopicResults(final CreateTopicsResult createTopicsResu
                         final Throwable cause = executionException.getCause();
                         if (cause instanceof TopicExistsException) {
                             lastErrorsSeenForTopic.put(topicName, cause);
-                            log.info("Internal topic {} already exists. Topic is probably marked for deletion. " +
-                                "Will retry to create this topic later (to let broker complete async delete operation first)",
-                                topicName);
+                            log.info("Could not create topic {}. Topic is probably marked for deletion (number of partitions is unknown).\n"
+                                    + "Will retry to create this topic in {} ms (to let broker finish async delete operation first).\n"
+                                    + "Error message was: {}",
+                                topicName, retryBackOffMs,
+                                cause.toString()
+                            );
                         } else if (cause instanceof TimeoutException) {
                             lastErrorsSeenForTopic.put(topicName, cause);
                             log.info("Creating internal topic {} timed out.", topicName);
                         } else {
                             cleanUpCreatedTopics(createdTopics);
                             log.error("Unexpected error during creation of internal topic: ", cause);
-                            throw new StreamsException(
-                                String.format("Could not create internal topic %s for the following reason: ", topicName),
-                                cause
-                            );
+                            if (cause instanceof UnsupportedVersionException) {
+                                final String errorMessage = cause.getMessage();
+                                if (errorMessage != null &&
+                                    errorMessage.startsWith("Creating topics with default partitions/replication factor "
+                                        + "are only supported in CreateTopicRequest version 4+")) {
+
+                                    throw new StreamsException(String.format(
+                                        "Could not create topic %s, because brokers don't support configuration "
+                                            + "replication.factor=-1. You can change the replication.factor config or "
+                                            + "upgrade your brokers to version 2.4 or newer to avoid this error.",
+                                        topicName),
+                                        cause
```

Comment:
nit formatting. `cause` is indented correctly, it's weird that it align to the arguments above that are not parameters of `StreamsException` (like `cause`). Better:
```
throw new StreamsException(
    String.format(
        "Could not create topic %s, because brokers don't support configuration "
            + "replication.factor=-1. You can change the replication.factor config or "
            + "upgrade your brokers to version 2.4 or newer to avoid this error.",
        topicName
    ),
    cause
);
```